crates/noirc_evaluator/src/ssa/anchor.rs

```diff
@@ -134,25 +140,34 @@ impl Anchor {
                     len
                 }
             };
-            self.mem_map.get_mut(&array_id).unwrap()[mem_idx].push_front((item_pos, id));
+            let arr = self.mem_map.get_mut(&array_id).unwrap();
+            if mem_idx > arr.len() {
+                return Err(RuntimeErrorKind::ArrayOutOfBounds {
+                    index: mem_idx as u128,
+                    bound: arr.len() as u128,
+                }
+                .into());
+            }
+            arr[mem_idx].push_front((item_pos, id));
```

Comment:
Can combine this with the check above, `if let Some(elem) = arr.get(mem_idx) { ... push_front ... } else { return Err(...) }`.

And perhaps I'm evil but instead of `return Err(...).into();` I will sometimes do `Err(...)?;` since `?` calls `.into()` as well. Feel free to keep it as is if you don't like this suggestion.

Reply:
Done